homeassistant/components/watchyourlan/coordinator.py

```diff
@@ -0,0 +1,41 @@
+"""Coordinator for the WatchYourLAN integration."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from httpx import ConnectError, HTTPStatusError
+from watchyourlanclient import WatchYourLANClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WatchYourLANUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Class to manage fetching data from the WatchYourLAN API."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the coordinator."""
+        # Set up a regular polling interval (e.g., every 60 seconds)
+        update_interval = timedelta(seconds=60)  # Poll every 60 seconds
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="WatchYourLAN",
+            update_interval=update_interval,  # Add polling interval here
+        )
+        self.api_url = config_entry.data[CONF_URL]
```

Comment:
```suggestion
        self.api_url = self.config_entry.data[CONF_URL]
```
Where do we use this variable more?

Reply:
It's only used in the api_client setup. No need for it to be here, really. I'll remove it.